File: src/main/java/com/vaadin/starter/bakery/ui/components/FormButtonsBar.java
package com.vaadin.starter.bakery.ui.components;

import com.vaadin.flow.component.Component;
import com.vaadin.flow.component.ComponentEvent;
import com.vaadin.flow.component.ComponentEventListener;
import com.vaadin.flow.component.DomEvent;
import com.vaadin.flow.component.Tag;
import com.vaadin.flow.component.dependency.HtmlImport;
import com.vaadin.flow.shared.Registration;

/**
 * Java wrapper of the polymer element `form-buttons-bar`
 */
@Tag("form-buttons-bar")
@HtmlImport("src/components/form-buttons-bar.html")
public class FormButtonsBar extends Component {

	public void setSaveText(String saveText) {
		getElement().setProperty("saveText", saveText == null ? "" : saveText);
	}

	public void setCancelText(String cancelText) {
		getElement().setProperty("cancelText", cancelText == null ? "" : cancelText);
	}

	public void setDeleteText(String deleteText) {
		getElement().setProperty("deleteText", deleteText == null ? "" : deleteText);
	}

	public void setSaveDisabled(boolean saveDisabled) {
		getElement().setProperty("saveDisabled", saveDisabled);
	}

	public void setCancelDisabled(boolean cancelDisabled) {
		getElement().setProperty("cancelDisabled", cancelDisabled);
	}

	public void setDeleteDisabled(boolean deleteDisabled) {
		getElement().setProperty("deleteDisabled", deleteDisabled);
	}

	@DomEvent("save")
	public static class SaveEvent extends ComponentEvent<FormButtonsBar> {
		public SaveEvent(FormButtonsBar source, boolean fromClient) {
			super(source, fromClient);
		}
	}

	public Registration addSaveListener(ComponentEventListener<SaveEvent> listener) {
		return addListener(SaveEvent.class, listener);
	}

	@DomEvent("cancel")
	public static class CancelEvent extends ComponentEvent<FormButtonsBar> {
		public CancelEvent(FormButtonsBar source, boolean fromClient) {
			super(source, fromClient);
		}
	}

	public Registration addCancelListener(ComponentEventListener<CancelEvent> listener) {
		return addListener(CancelEvent.class, listener);
	}

	@DomEvent("delete")
	public static class DeleteEvent extends ComponentEvent<FormButtonsBar> {
		public DeleteEvent(FormButtonsBar source, boolean fromClient) {
			super(source, fromClient);
		}
	}

	public Registration addDeleteListener(ComponentEventListener<DeleteEvent> listener) {
		return addListener(DeleteEvent.class, listener);
	}
}

File: frontend/src/components/form-buttons-bar.js
import { PolymerElement } from '@polymer/polymer/polymer-element.js';
import '@vaadin/vaadin-button/src/vaadin-button.js';
import './buttons-bar.js';
import { html } from '@polymer/polymer/lib/utils/html-tag.js';
{
  class FormButtonsBarElement extends PolymerElement {
    static get template() {
      return html`
<buttons-bar>
 <vaadin-button id="save" slot="left" theme="raised primary">
   Save
 </vaadin-button>
 <vaadin-button id="cancel" slot="left" theme="raised">
   Cancel
 </vaadin-button>
 <vaadin-button id="delete" slot="right" theme="raised tertiary error">
   Delete
 </vaadin-button>
</buttons-bar>
`;
    }

    static get is() {
      return 'form-buttons-bar';
    }
  }
  window.customElements.define(FormButtonsBarElement.is, FormButtonsBarElement);
}
